scrapy_poet/middleware.py

```diff
@@ -44,15 +49,23 @@ def process_response(self, request: Request, response: Response,
         its type. Otherwise, this middleware doesn't populate
         ``request.cb_kwargs`` for this argument.
 
-        .. warning::
+        Currently, we are able to inject instances of the following
+        classes as *provider* dependencies:
 
-            We should be able to inject any type into classes that inherit from
-            ``web_poet.pages.Injectable``, but currently, we're only able to
-            build and inject ``scrapy.Response`` instances.
+        - :class:`~.DummyResponse`
+        - :class:`~scrapy.http.Request`
+        - :class:`~scrapy.http.Response`
+        - :class:`~scrapy.settings.Settings`
         """
         # Find out the dependencies
         callback = utils.get_callback(request, spider)
-        plan, provider_instances = utils.build_plan(callback, response)
+        dependencies = {
+            Request: request,
+            Response: response,
+            Settings: spider.settings,
+            utils.DummyResponse: self.get_dummy_response(request),
```

Comment:
+1 to remove `DummyResponse`

Reply:
Well, I changed my opinion. See https://github.com/scrapinghub/scrapy-poet/pull/30#issuecomment-632580102